pkg/controller/workload/workload_event.go

```diff
@@ -331,15 +370,33 @@ func handleDataWithoutService(workload *workloadapi.Workload) error {
 		err error
 		bk  = BackendKey{}
 		bv  = BackendValue{}
+		fk  = FrontendKey{}
+		fv  = FrontendValue{}
 	)
 	bk.BackendUid = hashName.StrToNum(workload.GetUid())
 	ips := workload.GetAddresses()
 	for _, ip := range ips {
+		if waypoint := workload.GetWaypoint(); waypoint != nil {
+			addr := waypoint.GetAddress().Address
+			bv.WaypointAddr = nets.ConvertIpByteToUint32(addr)
+			bv.WaypointPort = waypoint.GetHboneMtlsPort()
+		}
+
 		bv.IPv4 = nets.ConvertIpByteToUint32(ip)
 		if err = BackendUpdate(&bk, &bv); err != nil {
 			log.Errorf("Update backend map failed, err:%s", err)
 			return err
 		}
+
+		// stored PodIP in the frontend map for Pod to Pod access.
+		// FrontendKey:{IPv4:<PodIP>, Port:0}, FrontendValue:{ServiceID:BackendUid}
+		fk.IPv4 = binary.LittleEndian.Uint32(ip)
+		fk.Port = 0
```

Comment:
What confuses me is for service, we  have multiports as keys. Why is it different for pod

Reply:
For pod access, firstly, the port is fixed and can be obtained from the input parameters of ebpf, secondly, the workload message  of a pod does not contain specific port information; For service access, it is necessary to find the corresponding target port based on the service port.